---
title: "100-basics"
output: html_document
date: "2023-07-25"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
```

```{r}
meta<-read.csv("meta/PUTSampleBarcodes_062623.csv") %>% separate(SampleName, into=c("Trib","Species","Method","Year","Number"), remove = FALSE) 
meta
```
```{r}
meta %>% group_by(PlateName) %>% summarize(Count=n())
```

Lots of samples over several years!         


Have raw data.  Looks like nine plates sequenced on two lanes. 
230720_HiSeq_Run751      
HS: PE 150+8+8+150      

Lane 1 Project_ASEF_BMAG077_L1         
PUT014RAD	CAAAAG
PUT015RAD	CAACTA
PUT016RAD	CAGGCG
PUT017RAD	CATGGC
PUT018RAD	CATTTT
PUT019RAD	CCAACA
PUT020RAD	ACAGTG
PUT021RAD	GCCAAT
PUT022RAD	TAATCG

Lane 2     Project_ASEF_BMAG077_L2


PUT014RAD	CAAAAG
PUT015RAD	CAACTA
PUT016RAD	CAGGCG
PUT017RAD	CATGGC
PUT018RAD	CATTTT
PUT019RAD	CCAACA
PUT020RAD	ACAGTG
PUT021RAD	GCCAAT
PUT022RAD	TAATCG

Putting in data/raw     

check md5, combine lanes.       
See 100.1-combine.sh     
module load parallel     
srun -p high -t 02:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=9 parallel -j 9 < 100.1-combine.sh    

Set up some scripts to demultiplex in scripts    

Checking barcodes `grep GGACAAGCTATGCAGG  --color PUT014RAD-R1.fastq `     

Looks like they're there!     

Rename
PUT014RAD_RA_GGAAACATCGTGCAGG.fastq
   Well          SampleName Trib Species Method Year Number PlateName PlateBarcodeName PlateBarcodeSeq SampleBarcode
1    A1  PC_CHN_CS_2020-020   PC     CHN     CS 2020    020 PUT017rad       NEBNext 34          CATGGC      ACAAGCTA
2    B1  PC_CHN_CS_2020-022   PC     CHN     CS 2020    022 PUT017rad       NEBNext 34          CATGGC      AACAACCA
3    C1  PC_CHN_CS_2020-023   PC     CHN     CS 2020    023 PUT017rad       NEBNext 34          CATGGC      AGATCGCA

```{r}
rename<-meta %>% as_tibble()
rename$Plate<-gsub("rad","",rename$PlateName)
forward<-rename %>% mutate(SeqFile=paste0(Plate,"RAD_RA_GG",SampleBarcode,"TGCAGG.fastq")) %>% dplyr::select(SeqFile,SampleName) %>%
  mutate(Command=paste0("mv data/raw/",SeqFile," data/split/",SampleName,"-R1.fastq"))
reverse<-rename %>% mutate(SeqFile=paste0(Plate,"RAD_RB_GG",SampleBarcode,"TGCAGG.fastq")) %>% dplyr::select(SeqFile,SampleName) %>%
  mutate(Command=paste0("mv data/raw/",SeqFile," data/split/",SampleName,"-R2.fastq"))

files<-bind_rows(forward,reverse) 
write_tsv(files %>% select(Command), col_names = FALSE, file="100.2-rename-fish.sh")
```

Moved and renamed   

(base) maccamp@farm:~/putah-creek/data/split$ ls | grep R1 | perl -pe 's/.fastq//g'  > forward
(base) maccamp@farm:~/putah-creek/data/split$ ls | grep R2 | perl -pe 's/.fastq//g' > reverse
(base) maccamp@farm:~/putah-creek/data/split$ ls | grep R1 | perl -pe 's/-R1.fastq//g' > names
(base) maccamp@farm:~/putah-creek/data/split$ paste forward reverse names > to-align.txt
bash ../../doAlign-rad-unzipped.sh to-align.txt /home/maccamp/genomes/chinook-2/GCF_018296145.1_Otsh_v2.0_genomic.fna 

counting up reads 
awk '{n++} END {print FILENAME "\t" n/4}' PC_CHN_CS_2020-024-R1.fastq
for f in *R1.fastq; do awk '{n++} END {print FILENAME "\t" n/4}' $f >> raw-counts.txt; done;

```{r}
counts<-meta %>% as_tibble() %>% mutate(SeqFile=paste0(SampleName,"-R1.fastq"))
raw<-read_tsv("outputs/100/raw-counts.txt", col_names = c("SeqFile","RawCounts")) %>% mutate(ReadCount=2*RawCounts) %>% select(-RawCounts)

counts <-counts %>% left_join(raw) %>% select(-SeqFile)


files<-list.files(path="outputs/100", patter="*.stats", full.names = TRUE)
reads<-lapply(files, read.csv, header=FALSE, col.names=c("SampleName","Aligned","Filtered","Coverage"))
reads<-bind_rows(reads)

counts<-left_join(counts,reads)

write_csv(counts, "meta/putah-creek-meta-with-counts.csv")

```

```{r}
ggplot(counts) +
  geom_histogram(aes(x=ReadCount, fill=Method), color="black") +
  scale_fill_viridis_d(option="plasma") +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  ylab("Count\n") +
  xlab("\nReads") +
  ggtitle("Raw Read Count") +
  theme(plot.title = element_text(hjust=0.5))

ggplot(counts) +
  geom_histogram(aes(x=Filtered, fill=Method), color="black") +
  scale_fill_viridis_d(option="plasma") +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  ylab("Count\n") +
  xlab("\nReads") +
  ggtitle("Filtered Read Count") +
  theme(plot.title = element_text(hjust=0.5))
```


```{r}
counts %>% filter(Filtered > 1e5) %>% group_by(Method, Year) %>% summarize(Count=n())
```
