---
title: "103-not-filtered"
output: html_document
date: "2023-08-09"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
```


```{r}
library(tidyverse)
library(RcppCNPy)
library(viridis)
```

```{r}
meta<-read_csv("meta/putah-creek-meta-with-counts.csv") %>% mutate(Path2=paste0("data/split/",SampleName,".sort.bam"))
meta %>% group_by(Method, Year) %>% summarise(Count=n())
```


Let's get a subset of fish and try some things. I suspect the CS samples to be problematic.    

```{r}
df1<-meta %>% filter(Reads > 5e5) 
df1 %>% group_by(Method, Year)  %>% mutate(Mean=mean(Filtered), Median=median(Filtered), Max=max(Filtered)) %>% select(Method, Year, Mean, Median, Max) %>%
  mutate(Count=n()) %>% unique() %>% arrange(Method,Year)
```
343
Can try df1 here....

```{r}
write_tsv(df1 %>% select(Path2), col_names = FALSE, file="bamlists/343.bamlist")
write_tsv(df1 %>% select(SampleName), col_names = FALSE, file="bamlists/343-sample-names.txt")

```



```{sh, eval=FALSE}
srun -p high -t 15:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 16  \
-bam bamlists/343.bamlist  -rf meta/chrom-list.txt \
-ref  ~/genomes/chinook-2/GCF_018296145.1_Otsh_v2.0_genomic.fna \
-minInd 172 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/103/343-ibs > outputs/103/343-ibs.out 2> outputs/103/343-ibs.err &

srun -p high -t 15:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 16  \
-bam bamlists/343.bamlist  -rf meta/chrom-list.txt \
-ref  ~/genomes/chinook-2/GCF_018296145.1_Otsh_v2.0_genomic.fna \
-minInd 257 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/103/343-ibs-75 > outputs/103/343-ibs-75.out 2> outputs/103/343-ibs-75.err &

srun -p high -t 15:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 16  \
-bam bamlists/343.bamlist  -rf meta/chrom-list.txt \
-ref  ~/genomes/chinook-2/GCF_018296145.1_Otsh_v2.0_genomic.fna \
-minInd 308 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/103/343-ibs-90 > outputs/103/343-ibs-90.out 2> outputs/103/343-ibs-90.err &

## GLS
srun -p high -t 15:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 16  \
-bam bamlists/343.bamlist -rf meta/chrom-list.txt \
-ref ~/genomes/chinook-2/GCF_018296145.1_Otsh_v2.0_genomic.fna \
-minInd 257 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-minMaf 0.05 \
-out outputs/103/343-gl-75 > outputs/103/343-gl-75.out 2> outputs/103/343-gl-75.err &
```

Calls

```{sh, eval=FALSE}
srun -t 24:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 309 -bam bamlists/343.bamlist  -rf meta/chrom-list.txt \
-ref  ~/genomes/chinook-2/GCF_018296145.1_Otsh_v2.0_genomic.fna \
-minMaf 0.05 -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2 -out outputs/103/343-snps > outputs/103/343-snps.out 2> outputs/103/343-snps.err &

srun -t 24:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 257 -bam bamlists/343.bamlist  -rf meta/chrom-list.txt \
-ref  ~/genomes/chinook-2/GCF_018296145.1_Otsh_v2.0_genomic.fna \
-minMaf 0.05 -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2 -out outputs/103/343-snps-75 > outputs/103/343-snps-75.out 2> outputs/103/343-snps75.err &
```
