---
title: "104-separate-calls"
output: html_document
date: "2023-08-17"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
```

Let's separate Carcass Samples from others and get calls and find overlaps     

```{r}
meta<-read_csv("meta/putah-creek-meta-with-counts.csv") %>% mutate(Path=paste0("data/split/",SampleName,".sort.flt.bam"))
meta %>% group_by(Method, Year) %>% summarise(Count=n())
```
_1_ Carcass Survey      

```{r}
cs<-meta %>% filter(Method=="CS") %>% filter(Filtered > 1.5e5)
cs %>% summarise(Count=n(), Mean=mean(Filtered), Median=median(Filtered)) 

ggplot(cs) +
  geom_histogram(aes(x=Filtered)) +
  xlim(0, max(cs$Filtered)+1e5) +
  theme_bw() 
```

```{r}
write_tsv(cs %>% select(Path), col_names = FALSE, file="bamlists/cs17.bamlist")
```

```{sh, eval=FALSE}
srun -t 2:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 15 -bam bamlists/cs17.bamlist  -rf meta/chrom-list.txt \
-ref  ~/genomes/chinook-2/GCF_018296145.1_Otsh_v2.0_genomic.fna \
-minMaf 0.05 -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2 -out outputs/104/cs-snps > outputs/104/cs17-snps.out 2> outputs/104/cs17-snps.err &
```
 106 snps
 
 
```{r}
rst<-meta %>% filter(Method=="RST") %>% filter(Filtered > 3.4e5) %>% filter(Filtered< 4.5e5)
rst %>% summarise(Count=n(), Mean=mean(Filtered), Median=median(Filtered)) 

write_tsv(rst %>% select(Path), col_names = FALSE, file="bamlists/rst20.bamlist")

``` 

```{sh, eval=FALSE}
srun -t 2:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 18 -bam bamlists/rst20.bamlist  -rf meta/chrom-list.txt \
-ref  ~/genomes/chinook-2/GCF_018296145.1_Otsh_v2.0_genomic.fna \
-minMaf 0.05 -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2 -out outputs/104/rst-snps > outputs/104/rst20-snps.out 2> outputs/104/rst20-snps.err &
```

336 snps here.

```{r}
css<-read_tsv("outputs/104/cs-snps.geno.gz", col_names = FALSE) 
css<-css %>% select(X1,X2) %>% rename(Chrom=X1, Pos=X2) %>% mutate(Site=paste0(Chrom,"-",Pos))

rss<-read_tsv("outputs/104/rst-snps.geno.gz", col_names = FALSE) 
rss<-rss %>% select(X1,X2) %>% rename(Chrom=X1, Pos=X2)  %>% mutate(Site=paste0(Chrom,"-",Pos))

```

```{r}
shared<-css %>% filter(Site %in% rss$Site)
shared
write_tsv(shared %>% select(Chrom, Pos), col_names = FALSE, file="meta/sites-63.tsv")
```

What about the set of 243 individuals ? 
Combining with chroms 
(base) Macs-MacBook-Pro-2:meta mac$ cut -f 1 sites-63.tsv | sort | uniq > sites-63-chroms.tsv


```{sh, eval=FALSE}
srun -t 3:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-bam bamlists/243.bamlist  -rf meta/sites-63-chroms.tsv -sites meta/sites-63.tsv \
-ref  ~/genomes/chinook-2/GCF_018296145.1_Otsh_v2.0_genomic.fna \
-minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2 -out outputs/104/243-snps > outputs/104/243-snps.out 2> outputs/104/243-snps.err &
```
