---
title: "105-likely-combos"
output: html_document
date: "2023-08-29"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message=FALSE)
```

I imagine the 2017 Carcass surveys produce RST-2018 fish, maybe that is true. 

```{r}
library(tidyverse)
library(vcfR)
library(adegenet)
library(poppr)
```


```{r}
meta<-read_csv("meta/putah-creek-meta-with-counts.csv") %>% mutate(Path=paste0("data/split/",SampleName,".sort.flt.bam")) %>%
  mutate(Pop=paste0(Method,"-",Year))
meta %>% group_by(Method, Year,Pop) %>% summarise(Count=n())
```

```{r}
meta %>% filter(Filtered > 1e5) %>% group_by(Method, Year,Pop) %>% summarise(Count=n())
```

```{r}
ggplot(meta %>% filter(Filtered > 1e5)) +
  geom_histogram(aes(x=Filtered, fill=Pop)) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="H") +
  ylab("Count\n") +
  xlab("Filtered Read Count")
```

```{r}
m124<-meta %>% filter(Pop %in% c("CS-2017","RST-2018")) %>% filter(Filtered > 1e5)
m124 %>% group_by(Method, Year,Pop) %>% summarise(Count=n())
write_csv(m124, file="meta/124.csv")
```


```{r}
ggplot(m124) +
  geom_histogram(aes(x=Filtered, fill=Pop)) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="H") +
  ylab("Count\n") +
  xlab("Filtered Read Count")
```


Set up for calls 
```{r}
m124 %>% select(Path) %>% write_tsv("bamlists/124.bamlist", col_names = FALSE)
```
-rf meta/chrom-list.txt

```{sh, eval=FALSE}
srun -t 2:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  \
$HOME/angsd/angsd -nthreads 24 \
-minInd 93 -bam bamlists/124.bamlist -r NC_056429.1  \
-ref  ~/genomes/chinook-2/GCF_018296145.1_Otsh_v2.0_genomic.fna \
-minMaf 0.05 -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 4 -doPost 1 -postCutoff 0.9 -doPlink 2 -out outputs/105/124-snps-75 > outputs/105/124-snps-75.out 2> outputs/105/124-snps-75.err &
```


