---
title: "105-likely-combos"
output: html_document
date: "2023-08-29"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message=FALSE)
```

I imagine the 2017 Carcass surveys produce RST-2018 fish, maybe that is true. 

```{r}
library(tidyverse)
library(vcfR)
library(adegenet)
library(poppr)
```


```{r}
meta<-read_csv("meta/putah-creek-meta-with-counts.csv") %>% mutate(Path=paste0("data/split/",SampleName,".sort.flt.bam")) %>%
  mutate(Pop=paste0(Method,"-",Year))
meta %>% group_by(Method, Year,Pop) %>% summarise(Count=n())
```

```{r}
meta %>% filter(Filtered > 1e5) %>% group_by(Method, Year,Pop) %>% summarise(Count=n())
```

```{r}
ggplot(meta %>% filter(Filtered > 1e5)) +
  geom_histogram(aes(x=Filtered, fill=Pop)) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="H") +
  ylab("Count\n") +
  xlab("Filtered Read Count")
```

```{r}
m124<-meta %>% filter(Pop %in% c("CS-2017","RST-2018")) %>% filter(Filtered > 1e5)
m124 %>% group_by(Method, Year,Pop) %>% summarise(Count=n())
write_csv(m124, file="meta/124.csv")
```


```{r}
ggplot(m124) +
  geom_histogram(aes(x=Filtered, fill=Pop)) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="H") +
  ylab("Count\n") +
  xlab("Filtered Read Count")
```


Set up for calls 
```{r}
m124 %>% select(Path) %>% write_tsv("bamlists/124.bamlist", col_names = FALSE)
```
-rf meta/chrom-list.txt

```{sh, eval=FALSE}
srun -t 2:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  \
$HOME/angsd/angsd -nthreads 24 \
-minInd 93 -bam bamlists/124.bamlist -r NC_056430.1  \
-ref  ~/genomes/chinook-2/GCF_018296145.1_Otsh_v2.0_genomic.fna \
-minMaf 0.05 -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 4 -doPost 1 -postCutoff 0.9 -doPlink 2 -out outputs/105/124-snps-75 > outputs/105/124-snps-75.out 2> outputs/105/124-snps-75.err &
```

(0 sites with  chrom1)


```{r}
m78<-meta %>% filter(Pop %in% c("CS-2017","RST-2018")) %>% filter(Filtered > 1.5e5)
m78 %>% group_by(Method, Year,Pop) %>% summarise(Count=n())
write_csv(m78, file="meta/78.csv")
```


```{r}
m78 %>% select(Path) %>% write_tsv("bamlists/78.bamlist", col_names = FALSE)
```
-rf meta/chrom-list.txt

```{sh, eval=FALSE}
srun -t 2:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  \
$HOME/angsd/angsd -nthreads 24 \
-minInd 59 -bam bamlists/78.bamlist -r NC_056430.1  \
-ref  ~/genomes/chinook-2/GCF_018296145.1_Otsh_v2.0_genomic.fna \
-minMaf 0.05 -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 4 -doPost 1 -postCutoff 0.9 -doPlink 2 -out outputs/105/78-snps-75 > outputs/105/78-snps-75.out 2> outputs/105/78-snps-75.err &
```

(0 on chrom1)
(0 on chrom2)

```{r}
m88<-meta %>% filter(Method =="RST") %>% filter(Filtered > 2e5) %>% filter(Filtered < 5e5)
m88 %>% group_by(Method, Year,Pop) %>% summarise(Count=n())
write_csv(m88, file="meta/88.csv")
m88 %>% select(Path) %>% write_tsv("bamlists/88.bamlist", col_names = FALSE)

```

```{sh, eval=FALSE}
srun -t 2:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  \
$HOME/angsd/angsd -nthreads 24 \
-minInd 66 -bam bamlists/88.bamlist -r NC_056430.1  \
-ref  ~/genomes/chinook-2/GCF_018296145.1_Otsh_v2.0_genomic.fna \
-minMaf 0.05 -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 4 -doPost 1 -postCutoff 0.9 -doPlink 2 -out outputs/105/88-snps-75 > outputs/105/88-snps-75.out 2> outputs/105/88-snps-75.err &
```

13 sites at minMaf 0.05
1 site at minMaf 0.10

```{r}
m31<-meta %>% filter(Method =="RST") %>% filter(Filtered > 3e5) %>% filter(Filtered < 5e5)
m31 %>% group_by(Method, Year,Pop) %>% summarise(Count=n())
m31 %>% select(Path) %>% write_tsv("bamlists/31.bamlist", col_names = FALSE)

```

```{sh, eval=FALSE}
srun -t 2:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  \
$HOME/angsd/angsd -nthreads 24 \
-minInd 23 -bam bamlists/31.bamlist -r NC_056430.1  \
-ref  ~/genomes/chinook-2/GCF_018296145.1_Otsh_v2.0_genomic.fna \
-minMaf 0.10 -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 4 -doPost 1 -postCutoff 0.9 -doPlink 2 -out outputs/105/31-snps-75 > outputs/105/31-snps-75.out 2> outputs/105/31-snps-75.err &
```

maf 0.05 59 snps on one chrom 
maf 0.10 14 snps on one chroms