---
title: "106-ngs-relate"
output: html_document
date: "2023-08-30"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r}
library(tidyverse)
library(vcfR)
library(adegenet)
```


```{r}
meta<-read_csv("meta/putah-creek-meta-with-counts.csv") %>% mutate(Path=paste0("data/split/",SampleName,".sort.flt.bam")) %>%
  mutate(Pop=paste0(Method,"-",Year))
meta %>% group_by(Method, Year,Pop) %>% summarise(Count=n())
```



```{r}
m92<-meta %>% filter(Pop %in% c("CS-2017","RST-2018")) %>% filter(Filtered > 1e5) %>% filter(Filtered<2.5e5)
m92 %>% group_by(Method, Year,Pop) %>% summarise(Count=n())
```

```{r}
m92 %>% select(Path) %>% write_tsv(file="bamlists/92.bamlist", col_names = FALSE)
```


https://github.com/ANGSD/NgsRelate

```{sh, eval=FALSE}
### First we generate a file with allele frequencies (angsdput.mafs.gz) and a file with genotype likelihoods (angsdput.glf.gz).
./angsd -b filelist -gl 2 -domajorminor 1 -snp_pval 1e-6 -domaf 1 -minmaf 0.05 -doGlf 3

srun -t 4:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  \
$HOME/angsd/angsd -nthreads 24 \
-bam bamlists/92.bamlist -rf meta/chrom-list.txt \
-minMapQ 10 -minQ 20 \
-GL 2 -doMajorMinor 1 -SNP_pval 1e-6 -doMaf 1 -minMaf 0.05 -doGLF 3 \
-out outputs/106/angsdput > outputs/106/relate.out 2> outputs/106/relate.err &

### Then we extract the frequency column from the allele frequency file and remove the header (to make it in the format NgsRelate needs)

zcat outputs/106/angsdput.mafs.gz | cut -f5 | sed 1d > outputs/106/freq.frq

### run NgsRelate
# doing locally, -n is the number of samples

 ~/github/ngsRelate/ngsRelate  -g outputs/106/angsdput.glf.gz -n 92 -f outputs/106/freq.frq  -O outputs/106/newres
```
	-> Number of sites retained after filtering: 111590      
	

compiled ngsRelate locally    

"The first two columns contain indices of the two individuals used for the analysis. The third column is the number of genomic sites considered. The following nine columns are the maximum likelihood (ML) estimates of the nine jacquard coefficients, where K0==J9; K1==J8; K2==J7 in absence of inbreeding. Based on these Jacquard coefficients, NgsRelate calculates 11 summary statistics:"

rab is the pairwise relatedness


```{r}
rel<-read_tsv(file = "outputs/106/newres")
rel

write_tsv(rel, file="outputs/106/rel-92.csv")
```

```{r}
ggplot(rel) +
  geom_histogram(aes(x=theta))
```

```{r}
ggplot(rel %>%  filter(theta > 0.04)) +
  geom_histogram(aes(x=theta))
```


More fish!

```{r}
m196<-meta %>% filter(Filtered > 1e5) %>% filter(Filtered<3e5)
m196 %>% group_by(Method, Year,Pop) %>% summarise(Count=n())

m196 %>% select(Path) %>% write_tsv(file="bamlists/196.bamlist", col_names = FALSE)
```

```{sh, eval=FALSE}
srun -t 4:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  \
$HOME/angsd/angsd -nthreads 24 \
-bam bamlists/196.bamlist -rf meta/chrom-list.txt \
-minMapQ 10 -minQ 20 \
-GL 2 -doMajorMinor 1 -SNP_pval 1e-6 -doMaf 1 -minMaf 0.05 -doGLF 3 \
-out outputs/106/angsdput-196 > outputs/106/relate-196.out 2> outputs/106/relate-196.err &

### Then we extract the frequency column from the allele frequency file and remove the header (to make it in the format NgsRelate needs)

zcat outputs/106/angsdput-196.mafs.gz | cut -f5 | sed 1d > outputs/106/freq-196.frq

### run NgsRelate
# doing locally, -n is the number of samples

 ~/github/ngsRelate/ngsRelate  -g outputs/106/angsdput-196.glf.gz -n 196 -f outputs/106/freq-196.frq  -O outputs/106/newres-196.tsv
```

	-> Frequency file: 'outputs/106/freq-196.frq' contain 151512 number of sites


```{r}
rel<-read_tsv(file = "outputs/106/newres-196.tsv")
rel

write_tsv(rel, file="outputs/106/rel-196.csv")
```


```{r}
ggplot(rel) +
  geom_histogram(aes(x=theta))
```


```{r}
ggplot(rel %>%  filter(theta > 0.04)) +
  geom_histogram(aes(x=theta))
```


0.125 should be something


```{r}
rel %>% filter(theta > 0.12) 
```

Getting sample names
```{r}
as<-m196 %>% select(SampleName) %>% mutate(a=1:n()-1) %>% rename(SampleA=SampleName)
bs<-m196 %>% select(SampleName) %>% mutate(b=1:n()-1) %>% rename(SampleB=SampleName)
comb<-left_join(rel,as) %>% left_join(bs) %>% relocate(SampleA, SampleB)
write_csv(comb %>% select(SampleA, SampleB, nSites, theta, rab, Fa, Fb, fraternity) %>% arrange(-theta), file="outputs/106/rel-196.csv")

comb %>% select(SampleA, SampleB, nSites, theta, rab, Fa, Fb, fraternity) %>% arrange(-theta) %>% filter( theta>0.1)

```

```{r}
ggplot(comb) + 
  geom_point(aes(x=R1, y=R0))

```

```{r}
ggplot(comb) + 
  geom_point(aes(x=R1, y=KING))

```

```{r}
comb %>% filter(theta > 0.125) %>% arrange(-rab) %>% select(rab,theta, SampleA, SampleB, nSites) 
```

## 243 fish


```{sh, eval=FALSE}
srun -t 4:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  \
$HOME/angsd/angsd -nthreads 24 \
-bam bamlists/243.bamlist -rf meta/chrom-list.txt \
-minMapQ 10 -minQ 20 \
-GL 2 -doMajorMinor 1 -SNP_pval 1e-6 -doMaf 1 -minMaf 0.05 -doGLF 3 \
-out outputs/106/angsdput-243 > outputs/106/relate-243.out 2> outputs/106/relate-243.err &

### Then we extract the frequency column from the allele frequency file and remove the header (to make it in the format NgsRelate needs)

zcat outputs/106/angsdput-243.mafs.gz | cut -f5 | sed 1d > outputs/106/freq-243.frq

### run NgsRelate
# doing locally, -n is the number of samples

 ~/github/ngsRelate/ngsRelate  -g outputs/106/angsdput-243.glf.gz -n 243 -f outputs/106/freq-243.frq  -O outputs/106/newres-243.tsv
```

	-> Frequency file: 'outputs/106/freq-243.frq' contain 171816 number of sites    


```{r}
rel<-read_tsv(file = "outputs/106/newres-243.tsv")
rel

write_tsv(rel, file="outputs/106/rel-243.csv")
```


```{r}
ggplot(rel) +
  geom_histogram(aes(x=theta))
```


```{r}
ggplot(rel %>%  filter(theta > 0.06)) +
  geom_histogram(aes(x=theta))
```
```{r}
ggplot(rel %>%  filter(rab > 0.06)) +
  geom_histogram(aes(x=rab))

```


Getting sample names
```{r}
m243<-filter(meta, Filtered > 1e5)
as<-m243 %>% select(SampleName) %>% mutate(a=1:n()-1) %>% rename(SampleA=SampleName)
bs<-m243 %>% select(SampleName) %>% mutate(b=1:n()-1) %>% rename(SampleB=SampleName)
comb<-left_join(rel,as) %>% left_join(bs) %>% relocate(SampleA, SampleB)
write_csv(comb %>% select(SampleA, SampleB, nSites, theta, rab, Fa, Fb, theta, fraternity) %>% arrange(-theta), file="outputs/106/rel-243.csv")

```

```{r}
ggplot(comb) + 
  geom_point(aes(x=R1, y=R0)) +
  geom_hline(yintercept = 0, linetype=2)
```

```{r}
ggplot(comb) + 
  geom_point(aes(x=R1, y=KING)) +
  geom_hline(yintercept = 0, linetype=2)
```

```{r}
comb %>% filter(R1 > 0.25) %>% relocate(theta, rab)
```


```{r}
comb %>% filter(rab > 0.1) %>% arrange(-rab) %>% select(rab,theta, SampleA, SampleB, nSites)


```

```{r}
comb %>% select(SampleA, SampleB, nSites, rab,theta) %>%
  filter(SampleA %in% c("PC_CHN_CS_2017_142A","PC_CHN_CS_2017_162A","PC_CHN_CS_2017_233A","PC_CHN_CS_2017_129D")) %>% arrange(-rab)
```


```{r}
comb %>% filter(SampleA == "PC_CHN_RST_2018_432") %>% arrange(-rab) %>% relocate(rab, theta)
comb %>% filter(SampleB == "PC_CHN_RST_2018_432") %>% arrange(-rab) %>% relocate(rab, theta)

```

	PC_CHN_RST_2018_2483
```{r}

comb %>% filter(SampleA == "PC_CHN_RST_2021_2462") %>% arrange(-theta) %>% relocate(rab, theta)
comb %>% filter(SampleB == "PC_CHN_RST_2021_2462") %>% arrange(-theta) %>% relocate(rab, theta)

```


## Compare to Colony
PC_CHN_RST_2018_326	PC_CHN_CS_2017_140C	0.9972
PC_CHN_RST_2018_576	PC_CHN_CS_2017_140C	0.9972
PC_CHN_RST_2018_981	PC_CHN_CS_2017_140C	0.9972
PC_CHN_RST_2018_1445	PC_CHN_CS_2017_127D	0.9997
PC_CHN_RST_2018_1880	PC_CHN_CS_2017_127D	0.9997

```{r}
comb %>% filter(SampleA == "PC_CHN_CS_2017_140C") %>% filter(SampleB %in% c("PC_CHN_RST_2018_326","PC_CHN_RST_2018_576","PC_CHN_RST_2018_981")) %>% relocate(rab, theta) %>% arrange(-theta)
```

```{r}
comb %>% filter(SampleB == "PC_CHN_CS_2017_140C") %>% filter(SampleA %in% c("PC_CHN_RST_2018_326","PC_CHN_RST_2018_576","PC_CHN_RST_2018_981")) %>% relocate(rab, theta) %>% arrange(-theta)
```